### PR TITLE
[ZT] Add Gateway policy examples

### PR DESCRIPTION
**Pull request by maxvp**
- Moved API examples from `/cloudflare-one/examples/` to their own product directories in `/cloudflare-one/api-terraform/`.
- Added optional `directory` parameter to `list-examples` shortcode to allow tutorial locations other than `/<product>/examples/`.

### CODE REVIEW

File: layouts/shortcodes/list-examples.html [hunk 1]
```diff
@@ -1,6 +1,12 @@
 {{- $target := printf "%sexamples/" .Page.FirstSection.RelPermalink -}}
 {{- $pages := (where .Site.Pages "Section" .Page.Section) -}}
 
+{{- with .Get "directory" -}}
+  {{- $target = . -}}
+{{- end -}}
+
+{{- $target -}}
```
Comment (kodster28) [inline layouts/shortcodes/list-examples.html]:
```suggestion
```

**Review by kodster28 — APPROVED** [review]

File: content/cloudflare-one/api-terraform/_index.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ weight: 8
 
 # API and Terraform
 
-This section covers a few common use cases with the API and Terraform to manage Cloudflare Access. For specific API documentation, please see [api.cloudflare.com](https://api.cloudflare.com/).
+This section covers a few common use cases with the API and Terraform to manage Cloudflare Zero Trust. For specific API documentation, please see [api.cloudflare.com](https://api.cloudflare.com/).
```
Comment (ranbel) [inline content/cloudflare-one/api-terraform/_index.md]:
```suggestion
This section covers a few common use cases with the API and Terraform to manage Cloudflare Zero Trust. For more information, refer to our [API documentation](https://developers.cloudflare.com/api/) and [Terraform reference guide](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs).
```

File: content/cloudflare-one/api-terraform/rule-api-examples/_index.md [hunk 3]
```diff
@@ -38,6 +38,4 @@ Access users can create policies, including individual rule blocks inside of gro
 
 ## Example rule configurations
 
-These are commonly used rule configurations.
-
-{{<list-examples>}}
+{{<list-examples directory="/cloudflare-one/api-terraform/rule-api-examples/">}}
```
Comment (ranbel) [inline content/cloudflare-one/api-terraform/rule-api-examples/_index.md:41]: For consistency, can we rename this folder to access-api-examples?

File: content/cloudflare-one/api-terraform/gateway-api-examples/_index.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+type: overview
+pcx_content_type: configuration
+title: Gateway API examples
+weight: 2
+layout: list
+---
+
+# Gateway API examples
+
+Gateway users can create DNS, network, and HTTP policies, including policies with multiple traffic, identity, and device posture conditions.
```
Comment (ranbel) [inline content/cloudflare-one/api-terraform/gateway-api-examples/_index.md]:
```suggestion
You can use the Cloudflare Gateway API to create DNS, network, and HTTP policies, including policies with multiple traffic, identity, and device posture conditions.
```

Comment (ranbel) [reply content/cloudflare-one/api-terraform/gateway-api-examples/_index.md]: Usually "users" = eyeballs / employees
Should also update the Access examples page.

**Review by ranbel — APPROVED** [review]

**Pull request merged**